source/common/runtime/runtime_impl.cc

```diff
@@ -469,12 +474,26 @@ void RtdsSubscription::onConfigUpdateFailed(Envoy::Config::ConfigUpdateFailureRe
 
 void RtdsSubscription::start() { subscription_->start({resource_name_}); }
 
-void RtdsSubscription::validateUpdateSize(uint32_t num_resources) {
-  if (num_resources != 1) {
+void RtdsSubscription::validateUpdateSize(uint32_t added_resources_num,
+                                          uint32_t removed_resources_num) {
+  if (added_resources_num + removed_resources_num != 1) {
     init_target_.ready();
-    throw EnvoyException(fmt::format("Unexpected RTDS resource length: {}", num_resources));
-    // (would be a return false here)
+    throw EnvoyException(fmt::format("Unexpected RTDS resource length, number of added recources "
+                                     "{}, number of removed recources {}",
+                                     added_resources_num, removed_resources_num));
+  }
+}
+
+void RtdsSubscription::onConfigRemoved(
+    const Protobuf::RepeatedPtrField<std::string>& removed_resources) {
+  if (removed_resources[0] != resource_name_) {
+    throw EnvoyException(fmt::format("Unexpected RTDS runtime (expecting {}): {}", resource_name_,
```

Comment:
I would make this error mention that this is during removal somehow, e.g. "Unexpected removal of unknown RTDS runtime layer {}, expected {}"